src/ratbag_emu/protocol/base.py

```diff
@@ -199,33 +197,38 @@ def _send_packets(self, packets, total):
             next_time += 1 / self.hw_profile.get_report_rate()
         s.run()
 
+    @property
+    def info(self):
+        return self._owner.info
 
-class BaseDevice(object):
-    def __init__(self, rdesc=None, info=None, name='Generic Device',
-                 shortname='generic'):
-        self.info = info
-        self.name = name
-        self.shortname = shortname
-        self.endpoints = {}
+    @property
+    def shortname(self):
+        return self._owner.shortname
 
-        self.endpoints[0] = SimulatedDevice(self, rdesc, info, name, shortname)
 
-        self.protocol = None
+class BaseDevice(object):
+    device_list = {
+        # Logitech
+        'logitech-g-pro-wireless': LogitechGProWirelessDevice
+    }
 
+    def __init__(self):
+        assert self.name, 'Name is missing'
+        assert self.shortname, 'Shortname is missing'
+        assert self.info, 'Bus info is missing'
+        assert self.rdescs, 'Report descriptor(s) is missing'
```

Comment:
You are preventing anyone to instantiate a `BaseDevice` class here, so you are making it abstract in a way.
But I think we should stick to the previous behaviour, with named parameters, because this will give you the python facilities to know when you forgot to update one parameter in a subclass.

Reply:
Sure. I added a `self.initialized` properly to avoid calling `super().__init__()` with the already initialized attributes.